File: packages/sdk/sdk-ruby/tests/team_models_test.rb
require "minitest/autorun"
require_relative "../lib/index"

class TeamModelsTest < Minitest::Test
  def test_list_team_models_returns_payload
    client = AIStatsSdk::AIStats.new(
      api_key: "test",
      enable_deprecation_warnings: false
    )

    calls = []
    client.raw_client.define_singleton_method(:request) do |method:, path:, query: nil, headers: nil, body: nil|
      calls << [method, path, query, headers, body]
      {
        "ok" => true,
        "limit" => 2,
        "models" => [{"id" => "openai/gpt-5-mini", "endpoints" => ["responses"]}]
      }
    end

    response = client.list_team_models("limit" => "2", "endpoints" => "responses")

    assert_equal true, response["ok"]
    assert_equal "openai/gpt-5-mini", response["models"][0]["id"]
    assert_equal [["GET", "/gateway/models/me", {"limit" => "2", "endpoints" => "responses"}, nil, nil]], calls
  end
end

File: packages/sdk/sdk-php/tests/team_models_test.php
<?php
declare(strict_types=1);

require_once __DIR__ . "/../src/index.php";

use AIStats\Sdk\AIStats;

function assert_true(bool $condition, string $message): void
{
    if (!$condition) {
        throw new RuntimeException($message);
    }
}

final class FakeTeamModelsClient extends \AIStats\Gen\Client
{
    public array $calls = [];

    public function __construct()
    {
        parent::__construct("https://example.test");
    }

    public function request(string $method, string $path, ?array $query = null, ?array $headers = null, $body = null)
    {
        $this->calls[] = [$method, $path, $query, $headers, $body];
        return [
            "ok" => true,
            "limit" => 2,
            "models" => [
                ["id" => "openai/gpt-5-mini", "endpoints" => ["responses"]],
            ],
        ];
    }
}

$client = new AIStats(
    apiKey: "test",
    basePath: "https://api.phaseo.app/v1",
    enableDeprecationWarnings: false
);

$fake = new FakeTeamModelsClient();
$ref = new ReflectionClass($client);
$prop = $ref->getProperty("client");
$prop->setAccessible(true);
$prop->setValue($client, $fake);

$response = $client->listTeamModels(["limit" => "2", "endpoints" => "responses"]);
assert_true(($response["ok"] ?? null) === true, "expected ok payload");
assert_true(($response["models"][0]["id"] ?? null) === "openai/gpt-5-mini", "expected team model id");
assert_true($fake->calls === [
    ["GET", "/models", ["limit" => "2", "endpoints" => "responses"], null, null],
], "expected request call for team models discovery");

echo "php team models tests ok" . PHP_EOL;
